core/src/main/java/org/elasticsearch/repositories/fs/FsRepository.java

```diff
@@ -54,10 +54,10 @@
         new Setting<>("location", "", Function.identity(), Property.NodeScope);
     public static final Setting<String> REPOSITORIES_LOCATION_SETTING =
         new Setting<>("repositories.fs.location", LOCATION_SETTING, Function.identity(), Property.NodeScope);
-    public static final Setting<ByteSizeValue> CHUNK_SIZE_SETTING =
-        Setting.byteSizeSetting("chunk_size", new ByteSizeValue(-1), Property.NodeScope);
-    public static final Setting<ByteSizeValue> REPOSITORIES_CHUNK_SIZE_SETTING =
-        Setting.byteSizeSetting("repositories.fs.chunk_size", new ByteSizeValue(-1), Property.NodeScope);
+    public static final Setting<ByteSizeValue> CHUNK_SIZE_SETTING = Setting.byteSizeSetting("chunk_size",
```

Comment:
Since the validation logic of the chunk_size setting depends on the repository implemention, we should also check in `BlobStoreRepository` that when `repository.chunkSize()` is not null then the `getBytes()` value is greater than 0?

Reply:
@tlrx Good point. I've pushed the fix for this. Could you take another look when you have a chance?

Reply:
@imotov Thanks! Sorry to nitpick but shouldn't we throw an IAE or something instead of just ignoring the partSize?

Reply:
Could you take another look?